chore(#1883): technical overview revamp (#1885)
medic/cht-docs: content/en/technical-overview/architecture/_index.md

---
title: "CHT Architecture"
linkTitle: "Architecture"

> weight: 1

Reviewer: ```suggestion
weight: 1
description: Components of the CHT Ecosystem
```

aliases:
   - /core
   - /core/overview/
---

> ### Components of the CHT Ecosystem

Reviewer: ```suggestion
```

Reviewer: I suggest we move this header to `description` so we can see it in the `relatedContent` pages. 

See the current reference without the `description`:

<img width="570" alt="Screenshot 2025-05-27 at 10 37 26" src="https://github.com/user-attachments/assets/a1e191b8-8466-4483-b332-6068ee78988d" />

{{< cards >}}
{{< card link="overview" title="Overview" subtitle="Overview of CHT components and their interactions" >}}

> {{< card link="cht-core" title="CHT-Core" subtitle="Central component providing authentication, role-based access control, and secure data endpoints." >}}

Reviewer: ```suggestion
{{< card link="cht-core" title="CHT Core" subtitle="Central component for building responsive, offline-first digital health apps" >}}
```

Reviewer: For consistency with the rest.

{{< card link="cht-conf" title="CHT Conf" subtitle="Command-line interface tool to manage and configure apps built using the CHT" >}}

> {{< card link="cht-conf" title="CHT Watchdog" subtitle="Dashboard visualization and alerting software" >}}

Reviewer: ```suggestion
{{< card link="cht-conf" title="CHT Watchdog" subtitle="Monitoring and alerting software" >}}
```

> {{< card link="cht-sync" title="CHT Sync" subtitle="Enables data synchronization between CouchDB and PostgreSQL for the purpose of analytics" >}}

Reviewer: ```suggestion
{{< card link="cht-sync" title="CHT Sync" subtitle="Data synchronization for the purpose of analytics" >}}
```

> {{< card link="cht-android" title="CHT Android" subtitle="Android wrapper for the CHT-Core Webapp" >}}

Reviewer: ```suggestion
{{< card link="cht-android" title="CHT Android" subtitle="Android wrapper for the CHT Core Webapp" >}}
```

> {{< card link="cht-interoperability" title="CHT Interoperability" subtitle="Exchanging data between the CHT and other health systems" >}}

Reviewer: ```suggestion
{{< card link="cht-interoperability" title="CHT Interoperability" subtitle="Standardized data exchange between the CHT and other health systems" >}}
```

> ---

Reviewer: nitpick: In the Hextra theme, we attempted to keep the section index pages lightweight by using `cards` or the `subPages` shortcode. I suggest separating this whole page into an `Overview`, and keeping the `_index.md` for the summary of the section.

Author: Ok, I changed this but am unsure this is the requested change. The _index page looks kinda empty now.

> ---

Reviewer: I suggest adding a CHT Android separate page where we could use elements from the [repo README](https://github.com/medic/cht-android) to explain the repo purpose and structure. 

Additionally, we might want to do the same for the `CHT Interoperability` product and add it to the architecture diagram.

Author: I agree about CHT-Android, but I wanted to keep this PR lightweight. I can add the page with text from the repo readme. 

Also agree about interop.

title: "Architecture of CHT Instances"
linkTitle: "Architecture"
weight: 1
aliases:
   - /core/overview/architecture
---

{{< hextra/hero-subtitle >}}
  Overview of CHT components and their interactions
{{< /hextra/hero-subtitle >}}

## Overview

<!-- make updates to this diagram on the google slides:            -->
<!-- https://docs.google.com/presentation/d/1j4jPsi-gHbiaLBfgYOyru1g_YV98PkBrx2zs7bwhoEQ/ -->

> {{< figure src="architecture.png" link="architecture.png" caption="Data Flows" >}}

Author: I played a bit with the diagram and did not get it to a place where I could say with confidence that I've improved it significantly. 

I agree with the less detailed diagrams!

> {{< figure src="architecture.png" link="architecture.png" caption="Data Flows" >}}

Reviewer 2: Wow! this is a big improvement - thank you!  

I suggest moving the left 3 boxes to be subgraphs.  I think this reduces cognitive load.   See below for what I mean.

Mermaid is notoriously very had to make look pretty (it's all function though!), but a nice to have would be to make all the boxes the same width.  I suspect this will be too hard though.  I have had luck targeting mermaid output with some very tactical CSS! again, just a nice to have.

![image](https://github.com/user-attachments/assets/fda5079d-b66b-4dd3-8734-8b2af5eb48e0)

(this was made by print page to PDF -> open in inkscape -> edit shapes in SVG -> screenshot :sweat_smile: )

The CHT Core Framework makes it faster to build full-featured, scalable digital health apps by providing a foundation developers can build on. These apps can support most languages, are [Offline-First]({{< ref "technical-overview/concepts/offline-first" >}}), and work on basic phones (via SMS), smartphones, tablets, and computers.

App developers are able to define health system roles, permissions and reporting hierarchies, and make use of five highly configurable areas of functionality: messaging, task and schedule management, decision support workflows, longitudinal person profiles, and analytics.

{{< see-also page="building" >}}

The Core Framework can be used to support the unique needs of a given health system and the work of community health workers, frontline supervisors, facility-based nurses, health system managers, and even patients and caregivers. Visit the Commuity Health Toolkit website to request a [self-guided demo](https://communityhealthtoolkit.org/contact) of an app built with the Core Framework. You can also explore an [overview of the Core Framework]({{< ref "features" >}}) and its key features.

> ### [cht-conf]({{< relref "technical-overview/architecture/cht-conf" >}})

Reviewer: ```suggestion
### [CHT Conf]({{< relref "technical-overview/architecture/cht-conf" >}})
```
nitpick: for consistency with the other CHT product names

A command-line interface tool to manage and configure CHT application.

### [CHT Sync]({{< relref "technical-overview/architecture/cht-sync" >}})

A suite of tools for extracting and normalizing data from the Core Framework's CouchDB, and rendering the data in analytics dashboards to visualize key data for a CHT deployment. Read more detail on the [CHT Sync overview page]({{< relref "technical-overview/architecture/cht-sync" >}}) and the [cht-sync GitHub repository](https://github.com/medic/cht-sync).

### [CHT Watchdog]({{< relref "technical-overview/architecture/cht-watchdog" >}})

> Monitoring and alerting for the CHT Core Framework to ensure CHWs are able to deliver care without interruption caused by server downtime.  Read more detail on the [CHT Watchdog overview page]({{< relref "technical-overview/architecture/cht-watchdog" >}}) and the [CHT Watchdog GitHub repository](https://github.com/medic/cht-watchdog).

Reviewer 2: ```suggestion
Monitoring and alerting for the CHT Core Framework to ensure CHWs are able to deliver care without interruption caused by server downtime.  Read more detail on the CHT Watchdog [technical overview page](/technical-overview/architecture/cht-watchdog) or the [GitHub repository](https://github.com/medic/cht-watchdog).
```
